imagedephi/gui.py

```diff
@@ -124,7 +125,54 @@ def image_url(path: str, key: str) -> str:
     )
 
 
-def get_image_response_from_ifd(ifd: IFD):
+def extract_thumbnail_from_image_bytes(ifd: IFD, file_name: str):
+    offsets = ifd["tags"][tifftools.Tag.TileOffsets.value]["data"]
+    byte_counts = ifd["tags"][tifftools.Tag.TileByteCounts.value]["data"]
+    num_tiles = len(offsets)
+
+    height = int(ifd["tags"][tifftools.Tag.ImageLength.value]["data"][0])
+    width = int(ifd["tags"][tifftools.Tag.ImageWidth.value]["data"][0])
+    samples = int(ifd["tags"][tifftools.Tag.SamplesPerPixel.value]["data"][0])
+    image_array = np.zeros((height, width, samples))
+    top: int = 0
+    left: int = 0
+
+    with open(file_name, "rb") as image_file:
+        for idx in range(num_tiles):
+            image_file.seek(int(offsets[idx]))
+            tile_bytes = BytesIO(image_file.read(int(byte_counts[idx])))
+            tile_image = Image.open(tile_bytes)
+            tile_array = np.array(tile_image)
+            tile_size = tile_image.size
+
+            bottom = top + tile_size[0]
+            right = left + tile_size[1]
+            if bottom > image_array.shape[0]:
+                bottom = image_array.shape[0]
+            if right > image_array.shape[1]:
+                right = image_array.shape[1]
+
+            x_width = bottom - top
+            y_length = right - left
+            image_array[top:bottom, left:right, :] = tile_array[0:x_width, 0:y_length, :]
```

Comment:
If this is the only thing we are using numpy for, we could have done this entirely with PIL.  But if numpy bundles properly for our executables, we should leave this as written.

Reply:
I was able to get rid of `numpy` in favor of `Image.paste`. See e28ffa6 for the changes.